testData/parsing/haxe/expressions/New.hx

```diff
@@ -0,0 +1,43 @@
+/* Tests parsing 'new' in a ternary expression */
+
+package test_tivo_229;
+
+class A {
+    public function new() {}
+}
+
+class B extends A {
+    public function new() { super(); }
+}
+
+class C extends B {
+    public function new( a:A ) { super(); }
+}
+
+class Test {
+    public function new() {}
+    
+    public function getA(want_a:Bool):A {
+        var qqq = (want_a == true) ? new A() : new B();
+        return qqq;
+    }
+
+    public function getB(want_a:Bool):A {
+        var qqq = null != new B() ? new A() : new B();
+        return qqq;
+    }
+
+    public function getC(want_a:Bool):A {
+        var qqq = null != new B() ? want_a == true ? new A() : new B() : want_a == false ? new A() : new B();
```

Comment:
I agree.  I had tried a couple of combinations and found more errors after I fixed the first one.  So I decided to try ternaries embedded in ternaries, and what do you know?  I got errors.  So this is actually a useful test case.